tests/test_routing.py

```diff
@@ -1022,6 +1032,221 @@ def test_host_named_repr() -> None:
     assert repr(route).startswith("Host(host='example.com', name='app', app=")
 
 
+@pytest.fixture
+def lifespan():
+    startup = MagicMock()
+    shutdown = MagicMock()
+
+    async def lifespan(scope: Scope, receive: Receive, send: Send):
+        assert scope["type"] == "lifespan"
+        assert scope["app"] == lifespan
+
+        message = await receive()
+        assert message["type"] == "lifespan.startup"
+        try:
+            startup()
+        except Exception:
+            await send(
+                {
+                    "type": "lifespan.startup.failed",
+                    "message": format_exc(),
+                }
+            )
+            return
+        await send({"type": "lifespan.startup.complete"})
+
+        message = await receive()
+        assert message["type"] == "lifespan.shutdown"
+        try:
+            shutdown()
+        except Exception:
+            await send(
+                {
+                    "type": "lifespan.shutdown.failed",
+                    "message": format_exc(),
+                }
+            )
+            return
+        await send({"type": "lifespan.shutdown.complete"})
```

Comment:
I'm curious what the rational was for implementing this here as opposed to just mounting a Starlette app within a Starlette app (`app = Starlette(routes=[Mount(app=Starlette(lifespan=...))], lifespan=...)`)

Reply:
Yeah I mainly, wanted to have a lifespan where I could easily overwrite the result of what happens during startup/shutdown. Might indeed be possible to simplify with a starlette app indeed.